airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/RedshiftDestination.kt

```diff
@@ -114,36 +121,126 @@ class RedshiftDestination :
+                    // when we perform final table related setup operations.
+                    // We just need the streamId to perform the calls in streamOperation.
+                    DestinationInitialStatus(
+                        streamConfig = streamConfig,
+                        isFinalTablePresent = false,
+                        initialRawTableStatus =
+                            InitialRawTableStatus(
+                                rawTableExists = false,
+                                hasUnprocessedRecords = true,
+                                maxProcessedTimestamp = Optional.empty(),
+                            ),
+                        isSchemaMismatch = true,
+                        isFinalTableEmpty = true,
+                        destinationState =
+                            RedshiftState(needsSoftReset = false, isAirbyteMetaPresentInRaw = true),
+                    ),
+                    FileUploadFormat.CSV,
+                    destinationColumns,
+                    disableTypeDedupe = true,
+                )
+            streamOperation.writeRecords(
+                streamConfig,
+                listOf(
+                        // Dummy message
+                        PartialAirbyteMessage()
+                            .withSerialized("""{"testKey": "testValue"}""")
+                            .withRecord(
+                                PartialAirbyteRecordMessage()
+                                    .withEmittedAt(System.currentTimeMillis())
+                                    .withMeta(
+                                        AirbyteRecordMessageMeta(),
+                                    ),
+                            )
+                    )
+                    .stream()
             )
-            RedshiftUtil.checkSvvTableAccess(database)
+            streamOperation.finalizeTable(
+                streamConfig,
+                StreamSyncSummary(recordsWritten = Optional.of(1)),
+            )
+
+            // And now that we have a table, simulate the next sync startup.
+            destinationHandler.gatherInitialState(listOf(streamConfig))
+            // (not bothering to verify the return value, maybe we should?)
+
+            // clean up the raw table, this is intentionally not part of actual sync code
+            // because we avoid dropping original tables directly.
+            destinationHandler.execute(
+                Sql.of(
+                    "DROP TABLE IF EXISTS \"${streamId.rawNamespace}\".\"${streamId.rawName}\";",
+                ),
+            )
+
             return AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED)
-        } catch (e: ConnectionErrorException) {
-            val message = getErrorMessage(e.stateCode, e.errorCode, e.exceptionMessage, e)
+        } catch (e: SQLException) {
+            // copied from AbstractJdbcDestination's attemptTableOperations
```

Comment:
we were already copying the `emitConfigErrorTrace` / `return AirbyteConnectionStatus` stuff from our superclass, so... I guess we'll copy the rest of it in here?